tests/io/eager/test_commit.py

```diff
@@ -171,6 +171,37 @@ def test_commit_dataset_from_dict(dataset_function, store):
     assert_frame_equal(df_expected, actual)
 
 
+def test_commit_dataset_from_nested_metapartition(store):
+    """
+    Check it is possible to use `commit_dataset` with nested metapartitions as input.
+    Original issue: https://github.com/JDASoftwareGroup/kartothek/issues/40
+    """
+
+    df = pd.DataFrame({"a": [1, 1, 2, 2], "b": [3, 4, 5, 6]})
+
+    create_empty_dataset_header(
+        store=store,
+        dataset_uuid="uuid",
+        table_meta={"table": make_meta(df, "table", ["a"])},
+        partition_on=["a"],
+    )
+
+    partitions = []
+    for x in range(2):
+        partitions.append(
+            write_single_partition(
+                store=store,
+                dataset_uuid="uuid",
+                data={"data": {"table": df}},
+                partition_on=["a"],
+            )
+        )
+
+    commit_dataset(
+        store=store, dataset_uuid="uuid", new_partitions=partitions, partition_on=["a"]
+    )
```

Comment:
please add a short assertion that the expected partitions are now part of the dataset.